rfd/0001-testing-guidelines.md

```diff
@@ -117,21 +117,21 @@ Subtests
 ```go
 func TestParseInt(t *testing.T) {
 	tests := []struct {
-		desc    string
-		in      string
-		want    int
-		wantErr bool
+		desc      string
+		in        string
+		want      int
+		assertErr assert.ErrorAssertionFunc
 	}{
-		{desc: "positive", in: "123", want: 123},
-		{desc: "negative", in: "-123", want: -123},
-		{desc: "non-numeric", in: "abc", wantErr: true},
-		{desc: "empty", in: "", wantErr: true},
+		{desc: "positive", in: "123", want: 123, assertErr: assert.NoError},
+		{desc: "negative", in: "-123", want: -123, assertErr: assert.NoError},
+		{desc: "non-numeric", in: "abc", assertErr: assert.Error},
+		{desc: "empty", in: "", assertErr: assert.Error},
```

Comment:
these are missing `want` field, for the `assert.Equal(t, got, tt.want)` check

Reply:
or does the `assert.Error` skips this check?

Reply:
Nice catch. I just realized that we should be using `testify/require` instead of `testify/assert`.
They have the same API but `require` terminates the test on failure, while `assert` lets it proceed.

With `require`, error is checked first and if that check fails, `want` is never checked.